src/main/java/apoc/path/PathExplorer.java

```diff
@@ -243,116 +261,49 @@ public static Traverser traverse(TraversalDescription traversalDescription,
-		}
-
-		td = td.uniqueness(uniqueness); // this is how Cypher works !! Uniqueness.RELATIONSHIP_PATH
-		// uniqueness should be set as last on the TraversalDescription
-		return td.traverse(startNodes);
-	}
-
-	// when no commas present, acts as a pathwide label filter
-	public static class LabelSequenceEvaluator implements Evaluator {
-		private List<LabelMatcherGroup> sequenceMatchers;
 
-		private Evaluation whitelistAllowedEvaluation;
-		private boolean endNodesOnly;
-		private boolean filterStartNode;
-		private boolean beginSequenceAtStart;
-		private long minLevel = -1;
+		if (nodeFilter != null && !nodeFilter.isEmpty()) {
+			List<Node> endNodes = nodeFilter.getOrDefault(END_NODES, Collections.EMPTY_LIST);
+			List<Node> terminatorNodes = nodeFilter.getOrDefault(TERMINATOR_NODES, Collections.EMPTY_LIST);
+			List<Node> blacklistNodes = nodeFilter.getOrDefault(BLACKLIST_NODES, Collections.EMPTY_LIST);
+			List<Node> whitelistNodes;
 
-		public LabelSequenceEvaluator(String labelSequence, boolean filterStartNode, boolean beginSequenceAtStart, int minLevel) {
-			List<String> labelSequenceList;
-
-			// parse sequence
-			if (labelSequence != null && !labelSequence.isEmpty()) {
-				labelSequenceList = Arrays.asList(labelSequence.split(","));
+			if (nodeFilter.containsKey(WHITELIST_NODES)) {
+				// need to add to new list since we may need to add to it later
+				// encounter "can't add to abstractList" error if we don't do this
+				whitelistNodes = new ArrayList<>(nodeFilter.get(WHITELIST_NODES));
 			} else {
-				labelSequenceList = Collections.emptyList();
+				whitelistNodes = Collections.EMPTY_LIST;
 			}
 
-			initialize(labelSequenceList, filterStartNode, beginSequenceAtStart, minLevel);
-		}
-
-		public LabelSequenceEvaluator(List<String> labelSequenceList, boolean filterStartNode, boolean beginSequenceAtStart, int minLevel) {
-			initialize(labelSequenceList, filterStartNode, beginSequenceAtStart, minLevel);
-		}
-
-		private void initialize(List<String> labelSequenceList, boolean filterStartNode, boolean beginSequenceAtStart, int minLevel) {
-			this.filterStartNode = filterStartNode;
-			this.beginSequenceAtStart = beginSequenceAtStart;
-			this.minLevel = minLevel;
-			sequenceMatchers = new ArrayList<>(labelSequenceList.size());
-
-			for (String labelFilterString : labelSequenceList) {
-				LabelMatcherGroup matcherGroup = new LabelMatcherGroup().addLabels(labelFilterString.trim());
-				sequenceMatchers.add(matcherGroup);
-				endNodesOnly = endNodesOnly || matcherGroup.isEndNodesOnly();
+			if (!blacklistNodes.isEmpty()) {
+				td = td.evaluator(NodeEvaluators.blacklistNodeEvaluator(blacklistNodes));
 			}
 
-			// if true for one matcher, need to set true for all matchers
-			if (endNodesOnly) {
-				for (LabelMatcherGroup group : sequenceMatchers) {
-					group.setEndNodesOnly(endNodesOnly);
-				}
+			Evaluator endAndTerminatorNodeEvaluator = NodeEvaluators.endAndTerminatorNodeEvaluator(endNodes, terminatorNodes);
```

Comment:
I think you can add multiple, individual evaluators, but then you are bound by the order and prioritization of the traversal-framework.